### PR TITLE
JP-1: feat: Create template

### PR DESCRIPTION
**Pull request by EinSev**
Create a template from a journey

<details>
<summary>Test Daten für User (da gerade keine User angezeigt werden wegen Username rework in User Settings</summary>

```
const users = ref([
    {
        id: "7A-6F-FF-06-C1-D8",
        firstName: "Cariotta",
        lastName: "Angrove",
        role: 2,
    },
    {
        id: "BD-F2-C4-18-03-DC",
        firstName: "Timofei",
        lastName: "Willerstone",
        role: 1,
    },
    {
        id: "AC-7E-F5-06-C2-D5",
        firstName: "Erich",
        lastName: "Gniewosz",
        role: 1,
    },
    {
        id: "1D-BC-51-D0-0A-E1",
        firstName: "Natala",
        lastName: "Smiley",
        role: 2,
    },
    {
        id: "4D-CC-64-F9-59-59",
        firstName: "Kial",
        lastName: "de Keep",
        role: 2,
    },
    {
        id: "A3-CB-34-B5-F2-B5",
        firstName: "Cayla",
        lastName: "Taye",
        role: 2,
    },
    {
        id: "E3-FF-FA-0B-20-4E",
        firstName: "Jaime",
        lastName: "Maestro",
        role: 2,
    },
    {
        id: "99-38-2C-CB-34-0E",
        firstName: "Rowland",
        lastName: "Collough",
        role: 2,
    },
    {
        id: "E1-8A-8C-72-DF-33",
        firstName: "Jonie",
        lastName: "Silbersak",
        role: 2,
    },
    {
        id: "8D-7B-74-45-78-DD",
        firstName: "Jo-anne",
        lastName: "Chinnock",
        role: 2,
    },
    {
        id: "C3-6F-6F-E1-8D-C5",
        firstName: "Suki",
        lastName: "McBrearty",
        role: 2,
    },
    {
        id: "11-4F-BD-6F-8C-EA",
        firstName: "Willamina",
        lastName: "Juleff",
        role: 1,
    },
    {
        id: "E5-5A-E7-C2-5F-B3",
        firstName: "Randi",
        lastName: "Leggott",
        role: 1,
    },
    {
        id: "64-A6-A8-81-6F-4C",
        firstName: "Gilbert",
        lastName: "McOnie",
        role: 1,
    },
    {
        id: "99-4A-0B-2F-2E-2B",
        firstName: "Sashenka",
        lastName: "Wagnerin",
        role: 1,
    },
    {
        id: "5D-0F-AA-3B-3F-8A",
        firstName: "Cyrillus",
        lastName: "Thatcher",
        role: 2,
    },
    {
        id: "6A-DA-7B-F2-5E-AD",
        firstName: "Timmi",
        lastName: "Stitson",
        role: 1,
    },
    {
        id: "86-71-00-39-23-33",
        firstName: "Cornie",
        lastName: "Vida",
        role: 1,
    },
    {
        id: "2D-F0-00-B2-8A-0F",
        firstName: "Janaya",
        lastName: "Sharphouse",
        role: 2,
    },
    {
        id: "11-AC-C6-18-94-86",
        firstName: "Keene",
        lastName: "Albasiny",
        role: 2,
    },
    {
        id: "AF-94-72-59-7E-66",
        firstName: "Osborne",
        lastName: "Van der Beek",
        role: 2,
    },
    {
        id: "2D-67-D9-4D-94-3C",
        firstName: "Marsiella",
        lastName: "Logg",
        role: 1,
    },
    {
        id: "DE-45-C3-D9-AB-92",
        firstName: "Gris",
        lastName: "Hawkshaw",
        role: 2,
    },
    {
        id: "4A-10-37-99-61-A8",
        firstName: "Sharity",
        lastName: "Hawson",
        role: 1,
    },
    {
        id: "43-C7-8E-61-BE-71",
        firstName: "Galvan",
        lastName: "Hainsworth",
        role: 1,
    },
    {
        id: "16-6D-81-8B-5B-D9",
        firstName: "Jilli",
        lastName: "Ciciotti",
        role: 2,
    },
    {
        id: "03-00-C0-83-1B-02",
        firstName: "Cosmo",
        lastName: "Dight",
        role: 1,
    },
    {
        id: "F9-E3-38-61-B9-A4",
        firstName: "Maryanne",
        lastName: "Nicholl",
        role: 2,
    },
    {
        id: "7D-52-3A-13-5B-9A",
        firstName: "Devin",
        lastName: "Mollitt",
        role: 2,
    },
    {
        id: "02-C4-7B-EC-FE-2D",
        firstName: "Tera",
        lastName: "Androli",
        role: 1,
    },
    {
        id: "CF-DD-53-09-D2-9E",
        firstName: "Evaleen",
        lastName: "Letts",
        role: 2,
    },
    {
        id: "D4-6A-F5-D0-6F-5B",
        firstName: "Quint",
        lastName: "Hynam",
        role: 2,
    },
    {
        id: "F8-7F-08-59-BD-08",
        firstName: "Emelia",
        lastName: "Guiett",
        role: 2,
    },
    {
        id: "9A-84-BC-8C-6C-42",
        firstName: "Leoline",
        lastName: "Densie",
        role: 1,
    },
    {
        id: "EA-94-C0-33-09-22",
        firstName: "Georgetta",
        lastName: "Pochon",
        role: 1,
    },
    {
        id: "B0-7A-F9-36-2A-EE",
        firstName: "Eugene",
        lastName: "Bolderstone",
        role: 1,
    },
    {
        id: "F2-27-5F-36-C2-27",
        firstName: "Fee",
        lastName: "Helm",
        role: 1,
    },
    {
        id: "D2-BB-57-B9-72-CF",
        firstName: "Doyle",
        lastName: "Faithfull",
        role: 2,
    },
    {
        id: "A9-30-87-31-FD-DE",
        firstName: "Bron",
        lastName: "Spellman",
        role: 1,
    },
    {
        id: "80-BB-5B-CD-D8-92",
        firstName: "Jerry",
        lastName: "Trussman",
        role: 1,
    },
    {
        id: "0A-EA-84-55-42-F7",
        firstName: "Syman",
        lastName: "Wythill",
        role: 2,
    },
    {
        id: "8F-FA-01-5C-71-5A",
        firstName: "Nels",
        lastName: "Nisot",
        role: 1,
    },
    {
        id: "FF-F1-87-18-EB-12",
        firstName: "Tedie",
        lastName: "Wiersma",
        role: 2,
    },
    {
        id: "DC-6A-04-31-EC-47",
        firstName: "Garwin",
        lastName: "Sayre",
        role: 2,
    },
    {
        id: "BE-F5-64-E6-27-B3",
        firstName: "Koralle",
        lastName: "Harley",
        role: 2,
    },
    {
        id: "53-24-15-1D-14-0A",
        firstName: "Harriet",
        lastName: "Goodrick",
        role: 2,
    },
    {
        id: "58-BB-55-69-E1-4B",
        firstName: "Ellyn",
        lastName: "Churly",
        role: 2,
    },
    {
        id: "FB-26-42-1D-C7-A8",
        firstName: "Jessie",
        lastName: "Furbank",
        role: 1,
    },
    {
        id: "4F-43-75-F6-11-EC",
        firstName: "Sarine",
        lastName: "Paffett",
        role: 2,
    },
    {
        id: "AD-1C-17-C8-D4-5E",
        firstName: "Dyanne",
        lastName: "Tilby",
        role: 1,
    },
    {
        id: "0B-61-65-8B-F0-06",
        firstName: "Justina",
        lastName: "Lehr",
        role: 1,
    },
    {
        id: "BE-F9-E3-65-F7-82",
        firstName: "Flossy",
        lastName: "Wittering",
        role: 1,
    },
    {
        id: "1E-D9-4A-47-82-F5",
        firstName: "Antoni",
        lastName: "Daugherty",
        role: 2,
    },
    {
        id: "C7-B6-41-9C-26-16",
        firstName: "Brendan",
        lastName: "Pittel",
        role: 1,
    },
    {
        id: "2A-E2-7E-57-C3-73",
        firstName: "Nora",
        lastName: "Gumb",
        role: 1,
    },
]);
```
</details>

### CODE REVIEW

File: frontend/components/TemplateDialog.vue [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                        class="mb-1 grid grid-cols-5 grid-rows-2 items-center xs:grid-cols-8 sm:grid-cols-4 sm:gap-x-20"
+                    >
+                        <label
+                            for="template-name"
+                            class="text-md col-start-1 row-start-1 font-medium md:text-lg"
+                        >
+                            <T key-name="form.input.template.name" />
+                        </label>
+                        <Field
+                            id="template-name"
```
Comment (stefll) [inline frontend/components/TemplateDialog.vue:181]: image
Bitte Abstand zwischen dem Input feld und "Templatename" verkleinern
Bitt die Errormessage linksbündig machen. Also Nachricht soll am Anfang dr input box beginnen

**Review by stefll — CHANGES_REQUESTED** [review]
Some change Review not full done

File: frontend/components/EditJourneyDialog.vue [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            close();
+            emit("deleteJourney", props.id);
+        },
+    });
+};
+
+/**
+ * form validation
+ * when submitting form, fields are checked for validation
+ */
```
Comment (stefll) [inline frontend/components/EditJourneyDialog.vue:71]: image
HandyVersion:
die untere zeile soll von dort beginnen wo die obere beginnt (orangener strich)
Wenn möglich Zeilenabstand kleiner machen (blauer Strich)

File: frontend/components/form/AddressInput.vue [hunk 3]
```diff
@@ -30,6 +30,7 @@ if (journey.getLat() && journey.getLong()) {
 }
```
Comment (stefll) [inline frontend/components/form/AddressInput.vue:31]: Error message shouldn't right after "create journey"- button is clicked. 
image

File: frontend/components/TemplateDialog.vue [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                toast.add({
+                    severity: "error",
+                    summary: t.value("common.toast.error.heading"),
+                    detail: t.value("common.error.unknown"),
+                    life: 6000,
+                });
+            }
+            savingTemplate.value = false;
+        },
+    });
```
Comment (stefll) [inline frontend/components/TemplateDialog.vue:118]: image
Error message is still there after putting in a name. It only disappears when clicking away from the input field.

File: frontend/pages/journey/[id].vue [hunk 5]
```diff
@@ -348,9 +549,13 @@ const handleUpload = (result: string) => {
                 <SvgDashboardIcon class="h-6 w-6" />
                 <p class="text-2xl hover:underline">Dashboard</p>
             </NuxtLink>
-            <SvgMenu
-                class="mx-5 h-8 w-8 hover:cursor-pointer md:mx-10 md:h-10 md:w-10"
-                @click="visibleSidebar = true"
+            <span
```
Comment (stefll) [inline frontend/pages/journey/[id].vue]: HAndyversion:
Remove the Word "Dashboard" and put the Dashboard sign to the left upper corner.

**Review by stefll — CHANGES_REQUESTED** [review]
Design Review: 2 Bugs and 2 design fixes

File: frontend/components/EditJourneyDialog.vue [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+const client = useSanctumClient();
+const confirm = useConfirm();
+
+const loadingEdit = ref(false);
+
+const close = () => {
+    emit("closeEditJourneyDialog");
+};
+
+const confirmDelete = (event: Event) => {
```
Comment (stefll) [inline frontend/components/EditJourneyDialog.vue:43]: Journey can be created even if destination was cleared

**Review by stefll — CHANGES_REQUESTED** [review]
Another bug

**Review by stefll — DISMISSED** [review]
Design Review: Done

File: frontend/components/ActivityMap.vue [hunk 7]
```diff
@@ -35,32 +38,48 @@ const activities = ref();
...
 function markerColor(activity: Activity) {
     return activity.calendar_activities?.length > 0
         ? colorAdded
         : colorNotAdded;
 }
 
-const lat = Number(journey.getLat());
-const long = Number(journey.getLong());
-const zoom = computed(() => ((long || lat) === null ? 1 : 8));
+const zoom = computed(() => ((long.value || lat) === null ? 1 : 8));
```
Comment (Ixpiria) [inline frontend/components/ActivityMap.vue]: please have a look at the coordinates here

File: frontend/components/EditJourneyDialog.vue [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+
+/**
+ * form validation
+ * when submitting form, fields are checked for validation
+ */
+const { handleSubmit } = useForm({
+    validationSchema: yup.object({
+        name: yup
+            .string()
+            .min(1, t.value("form.error.journey.name"))
```
Comment (Ixpiria) [inline frontend/components/EditJourneyDialog.vue]: is this necessary, if we have required too?

File: frontend/components/EditJourneyDialog.vue [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+            .required(t.value("form.error.journey.name"))
+            .matches(/^(?!\s+$).*/, t.value("form.error.journey.name"))
+            .label(t.value("form.input.journey.name")),
+        destination: yup
+            .string()
+            .min(1, t.value("form.error.journey.destination"))
```
Comment (Ixpiria) [inline frontend/components/EditJourneyDialog.vue]: here too

**Review by Ixpiria — APPROVED** [review]

**Review by stefll — APPROVED** [review]

**Pull request merged**